File: Example/XenditExampleObjC/XenditExampleObjC/NotoSerifFont.m
#import "NotoSerifFont.h"
#import <CoreText/CoreText.h>

@implementation NotoSerifFont

+ (void)register {
    NSArray<NSString *> *names = @[
        @"NotoSerif-Regular",
        @"NotoSerif-Medium",
        @"NotoSerif-SemiBold",
        @"NotoSerif-Bold",
    ];
    for (NSString *name in names) {
        NSURL *url = [[NSBundle mainBundle] URLForResource:name withExtension:@"ttf"];
        if (url) {
            CTFontManagerRegisterFontsForURL((__bridge CFURLRef)url, kCTFontManagerScopeProcess, NULL);
        } else {
            NSLog(@"[XenditExampleObjC] '%@.ttf' not found in bundle — add it to Copy Bundle Resources in Xcode.", name);
        }
    }
}

+ (UIFont *)regular  { return [UIFont fontWithName:@"NotoSerif-Regular"  size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightRegular]; }
+ (UIFont *)medium   { return [UIFont fontWithName:@"NotoSerif-Medium"   size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightMedium]; }
+ (UIFont *)semiBold { return [UIFont fontWithName:@"NotoSerif-SemiBold" size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightSemibold]; }
+ (UIFont *)bold     { return [UIFont fontWithName:@"NotoSerif-Bold"     size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightBold]; }

@end

File: Example/XenditExampleUIKit/XenditExampleUIKit/NotoSerifFont.swift
import CoreText
import UIKit

enum NotoSerifFont {

    static func register() {
        let names = [
            "NotoSerif-Regular",
            "NotoSerif-Medium",
            "NotoSerif-SemiBold",
            "NotoSerif-Bold",
        ]
        for name in names {
            guard let url = Bundle.main.url(forResource: name, withExtension: "ttf") else {
                print("[XenditExampleUIKit] '\(name).ttf' not found in bundle — add it to Copy Bundle Resources in Xcode.")
                continue
            }
            CTFontManagerRegisterFontsForURL(url as CFURL, .process, nil)
        }
    }

    static var regular:  UIFont { UIFont(name: "NotoSerif-Regular",  size: 14) ?? .systemFont(ofSize: 14, weight: .regular) }
    static var medium:   UIFont { UIFont(name: "NotoSerif-Medium",   size: 14) ?? .systemFont(ofSize: 14, weight: .medium) }
    static var semiBold: UIFont { UIFont(name: "NotoSerif-SemiBold", size: 14) ?? .systemFont(ofSize: 14, weight: .semibold) }
    static var bold:     UIFont { UIFont(name: "NotoSerif-Bold",     size: 14) ?? .systemFont(ofSize: 14, weight: .bold) }
}
